### ISSUE TITLE
How do I save the detected class names to a text file? 

### ISSUE THREAD

**Issue by NAyAyNA** [ultralytics/yolov5#9690]
I'm aware of the --save-txt. But it gives the class label along with the x,y co-ordinates.
I want just the class names.

**Comment by glenn-jocher** [comment]
@NAyAyNA 👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) models allow for simple model loading and inference in a pure python environment without using `detect.py`. 

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://docs.ultralytics.com/yolov5/tutorials/model_export) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
#                                            'custom', 'path/to/best.pt')  # custom model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
results.xyxy[0]  # im predictions (tensor)

results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie

results.pandas().xyxy[0].value_counts('name')  # class counts (pandas)
# person    2
# tie       1
```
<img width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">

See [YOLOv5 PyTorch Hub Tutorial](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) for details.

Good luck 🍀 and let us know if you have any other questions!

**Comment by NAyAyNA** [comment]
ok! But is there a way i can save the output that detect.py gives on the command line in a text file?

**Comment by glenn-jocher** [comment]
@NAyAyNA of course. results as a pandas dataframe is exportable anywhere: excel spreadsheets, CSV, text files, XML, JSON, etc.

**Comment by Symbadian** [comment]
> @NAyAyNA 👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) models allow for simple model loading and inference in a pure python environment without using `detect.py`.
> 
> ### Simple Inference Example
> This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://docs.ultralytics.com/yolov5/tutorials/model_export) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.
> 
> ```python
> import torch
> 
> # Model
> model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
> #                                            'custom', 'path/to/best.pt')  # custom model
> 
> # Images
> im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list
> 
> # Inference
> results = model(im)
> 
> # Results
> results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
> results.xyxy[0]  # im predictions (tensor)
> 
> results.pandas().xyxy[0]  # im predictions (pandas)
> #      xmin    ymin    xmax   ymax  confidence  class    name
> # 0  749.50   43.50  1148.0  704.5    0.874023      0  person
> # 2  114.75  195.75  1095.0  708.0    0.624512      0  person
> # 3  986.00  304.00  1028.0  420.0    0.286865     27     tie
> 
> results.pandas().xyxy[0].value_counts('name')  # class counts (pandas)
> # person    2
> # tie       1
> ```
> 
> <img alt="" width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">
> 
> See [YOLOv5 PyTorch Hub Tutorial](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) for details.
> 
> Good luck 🍀 and let us know if you have any other questions!

Hi @glenn-jocher,
Thanx for your work pal, really appreciate this loads!!!!

I have a question on this please, as I have been searching all over google and cannot seem to get a fruitful response.

**Can I get the same output using this command?**
```
python3 detect.py --weights /Users/yolov5/runs/train/exp6/weights/best.pt  --img 640 --conf 0.25 --iou-thres  0.10 --source '/Users/vids50.mp4' --line-thickness 1
```

**as demonstrate here:**
```
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie
```

*** IF yes!! may I request your guidance to this example, please?*** I've been struggling with this expression for some weeks now.

 I am getting the --saved txt files but not in the about format..please help
Thanx for acknowledging my digital requests.
